www/js/menu.js

```diff
@@ -314,6 +317,61 @@ const announcementSlideButton = h(
   ),
 );
 
+const createSyncModal = (content, code) => {
+  const syncModal = new tingle.Modal({
+    footer: true,
+    stickyFooter: false,
+    cssClass: ['sync-code-modal'],
+    closeMethods: ['overlay', 'button', 'escape'],
+  });
+  // open modal
+  syncModal.open();
+
+  if (code) {
+    syncModal.addFooterBtn('Copy', 'tingle-btn tingle-btn--primary', () => {
+      copy(code);
+      syncModal.close();
+      syncModal.destroy();
+    });
+  }
+
+  // set content
+  syncModal.setContent(content);
+  // add cancel button
+  const cancelTitle = 'Cancel';
+  syncModal.addFooterBtn(
+    cancelTitle,
+    'tingle-btn tingle-btn--pull-right tingle-btn--default',
+    () => {
+      syncModal.close();
+      syncModal.destroy();
+    },
+  );
+};
+
+const remoteSyncButton = h(
+  'li',
+  h(
+    'a.remote-sync-button',
+    {
+      onclick: async () => {
+        const onlineVal = await isOnline();
+        if (onlineVal) {
+          const code = await tryConnection();
+          if (code) {
+            document.getElementById('remote-sync-icon').style.color = 'green';
+            createSyncModal(`<h2>Sync code: ${code}</h2>`, code);
+          }
+        } else {
+          createSyncModal(`<h2>Must be online to use this feature.</h2>`, '');
```

Comment:
Do we even want to show sync button when user is offline?

Reply:
That is difficult since we don't poll or get an event. if the user is online and then loses the connection, we don't have a way of getting that info to hide/show the button. I think it better to check on click.